src/Sarif.Viewer.VisualStudio.Core/OutputWindowTracerListener.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.Sarif.Viewer
+{
+    public class OutputWindowTracerListener : TraceListener
+    {
+        private readonly string _name;
+        private readonly IVsOutputWindow _outputWindowService;
+
+        private IVsOutputWindowPane pane;
+
+        public OutputWindowTracerListener(IVsOutputWindow outputWindowService, string name)
+        {
+            this._outputWindowService = outputWindowService;
+            this._name = name;
+            Trace.Listeners.Add(this);
+        }
+
+        public override void Write(string message)
+        {
+            if (!SarifViewerPackage.IsUnitTesting)
+            {
+#pragma warning disable VSTHRD108 // Assert thread affinity unconditionally
+                ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
If we want to access the UI components in Visual Studio SDK, you have to make sure it happens on main UI thread to prevent VS UI frozen.

E.g. build will fail for below code with error message: 'error VSTHRD010: Accessing "IVsOutputWindowPane" should only be done on the main thread. Call Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread() first.'

```
        public override void Write(string message)
        {
            this.pane.OutputStringThreadSafe(message);
        }
```
Need to call VS SDK `ThreadHelper.ThrowIfNotOnUIThread()` for sync or `await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync()` for async call to avoid the build error. 
```
        public override void Write(string message)
        {
            ThreadHelper.ThrowIfNotOnUIThread();

            this.pane.OutputStringThreadSafe(message);
        }
```